rust/kernel/io_buffer.rs

```diff
@@ -93,10 +93,10 @@ pub trait IoBufferWriter {
     /// # Safety
     ///
     /// The input buffer must be valid.
-    unsafe fn write_raw(&mut self, data: *const u8, len: usize) -> KernelResult;
+    unsafe fn write_raw(&mut self, data: *const u8, len: usize) -> Result;
 
     /// Writes the contents of the given data into the io buffer.
-    fn write<T: WritableToBytes>(&mut self, data: &T) -> KernelResult<()> {
+    fn write<T: WritableToBytes>(&mut self, data: &T) -> Result<()> {
```

Comment:
We could clean all this up, while we are at it.

```suggestion
    fn write<T: WritableToBytes>(&mut self, data: &T) -> Result {
```

Reply:
Indeed. Fixed.